### PR TITLE
Revert LOD selection implementation to a previous working version

### LINKED ISSUE #78664: Directional Shadow Pancake Size breaks Automatic LOD
### Godot version

v4.1.beta3.official [ada712e06]

### System information

Godot v4.1.beta3 - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1050 Ti (NVIDIA; 31.0.15.3179) - Intel(R) Core(TM) i3-10105F CPU @ 3.70GHz (8 Threads)

### Issue description

Pancake Size changes Directional Shadow LOD very aggressively, not representative of the used resolution of the object in shadow map

### Steps to reproduce

Open The Project, and Play around with Pancake Size and observe Human Dummy Shadows going to hell

### Minimal reproduction project

[Pancake.zip](https://github.com/godotengine/godot/files/11859346/Pancake.zip)

### LINKED ISSUE #81995: LOD selection implementation is wrong
### Godot version

4.2.dev

### System information

n/a

## Issue

I've just come back to work on a project and I'm certain the LOD selection code is wrong after #67307
I've witnessed LOD selection artifacts even after #79590, and looking at the code changes from #67307 the implementation doesn't make sense.

Example of broken LOD selection viewing a MultiMeshInstance3D:
Viewed from one angle:
image
And with camera rotated just barely, clearly the wrong LOD is used:
image

The problematic code:
https://github.com/godotengine/godot/blame/26c4644b388afb775c0563e7f8d70a3215c1216b/servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp#L947-L967

```cpp
if (!inst->transformed_aabb.has_point(p_render_data->scene_data->cam_transform.origin)) {
	// Get the LOD support points on the mesh AABB.
	Vector3 lod_support_min = inst->transformed_aabb.get_support(p_render_data->scene_data->cam_transform.basis.get_column(Vector3::AXIS_Z));
	Vector3 lod_support_max = inst->transformed_aabb.get_support(-p_render_data->scene_data->cam_transform.basis.get_column(Vector3::AXIS_Z));

	// Get the distances to those points on the AABB from the camera origin.
	float distance_min = (float)p_render_data->scene_data->cam_transform.origin.distance_to(lod_support_min);
	float distance_max = (float)p_render_data->scene_data->cam_transform.origin.distance_to(lod_support_max);

	if (distance_min * distance_max < 0.0) {
		//crossing plane
		distance = 0.0;
	} else if (distance_min >= 0.0) {
		distance = distance_min;
	} else if (distance_max <= 0.0) {
		distance = -distance_max;
	}
}
```

### Implementation flaws

First of all, plainly using vector distance to support points will not work.

There are illogical situations where `distance_min` is greater than `distance_max`:
image

Additionally, `distance_min/max` will *never* be negative when calculated by `Vector3::distance_to`. Therefore the final `distance` will always be equal to `distance_min`, which can be arbitrarily far from the camera for long AABBs despite the mesh being so close to the camera.

**I recommend reverting this code** (I have a branch prepared) before trying to implement a better solution. A better solution may be harder to attain than you would expect (or prove me wrong).

### The purpose of LOD level selection

An ideal LOD selection should maintain optimal visual quality for all parts of the mesh (enforcing `projected triangle size * lod_bias < 1 pixel`). Therefore, you should locate the worst-case geometry on the mesh (the position closest to the camera view plane) and calculate the LOD based on the distance.
(This tracks with the orthogonal case which assigns `distance = 1.0` where the projected size is constant with respect to distance.)

### Identifying the actual problems
The discussion of whether LOD levels should be determined by radial distance or view-normal scalar projection #54885 is motivated by two core problems:
1. a need for frustum culling (if not already implemented)
2. LOD popping in/out with camera rotation

Solving problem 1 involves a different system than LODs, and problem 2 only exists if you decide to change `lod_bias`. The [documentation](https://docs.godotengine.org/en/stable/tutorials/3d/mesh_lod.html) suggests modifying the bias for performance, but I believe other approaches should take priority (mesh optimization, base model quality, etc.), otherwise LOD popping is completely unavoidable without implementing an independent solution. Even if rotational consistency is achieved, LODs will still pop in/out as the camera translates. This is the wrong place to solve the problem.

### Possible approach for rotationally-consistent LOD level selection

If we _need_ the marginal quality improvements from achieving rotational consistency, some notes:

1. Calculate shortest distance between AABB surface and the camera near plane (this is possibly non-trivial)
2. `distance *= 1.0 / cos(camera FOV angle from screen center to corner)` to make sure that LOD is high enough to ensure `projected triangle size * lod_bias < 1 pixel` is enforced in the corners of the screen

### Steps to reproduce

-

### Minimal reproduction project

-

### PR DESCRIPTION
**Pull request by briansemrau**
Fixes #81995

See issue description for reason for reverting

May also fix #78664
Relevant to #67479

### CODE REVIEW

File: drivers/gles3/rasterizer_scene_gles3.h [hunk 1]
```diff
@@ -113,6 +113,7 @@ struct RenderDataGLES3 {
 	int reflection_probe_pass = 0;
 
 	float lod_distance_multiplier = 0.0;
+	Plane lod_camera_plane = Plane();
```
Comment (AThousandShips) [inline drivers/gles3/rasterizer_scene_gles3.h:116]:
```suggestion
	Plane lod_camera_plane;
```
This is already defaulted

**Comment by briansemrau** [comment]
Failed check appears to be unrelated?

**Comment by mrjustaguy** [comment]
Resolves https://github.com/godotengine/godot/issues/78664 but not https://github.com/godotengine/godot/issues/67479

**Comment by fire** [comment]
I am ok with this, but I want the rendering team to take a look.

**Comment by akien-mga** [comment]
Superseded by #92290.

**Pull request closed without merging**